### PR TITLE
[Tests-Only] Add tests for gRPC

### LINKED ISSUE #29: write GRPC tests
see example https://github.com/owncloud/ocis-hello/blob/master/pkg/proto/v0/hello.pb.micro_test.go

### PR DESCRIPTION
**Pull request by dpakach**
Add gRPC integration tests for accounts and group related actions for
-  CreateAccount
- ListAccounts
- GetAccount
- UpdateAccount
- DeleteAccount
- ListGroup
- AddGroup
- DeleteGroup
- GetGroup
- UpdateGroup(not implemented)
- AddMember
- ListMembers
- RemoveMembers

fixes #29

### CODE REVIEW

**Comment by individual-it** [comment]
vet step failed for some reason :(
image

**Comment by dpakach** [comment]
> vet step failed for some reason :(
> image
Yes seems like somethings changed here, when I rebased it's not working now, I'll fix and push again.

**Comment by dpakach** [comment]
Looks like #30 changes a lot of APIs so waiting for that to get merged before proceeding with it.

**Comment by refs** [comment]
@dpakach indeed. I flagged #30 as `blocked` because of what you mentioned, it drastically changes the API and #22 will be merged first, then we will build on top of it for #30.

**Comment by butonic** [comment]
@dpakach we merged the initial group support. only updating groups is not yet covered. but tha api has settled for now.

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+	resp, err := listAccounts(t)
+	checkError(t, err)
+
+	assert.IsType(t, &proto.ListAccountsResponse{}, resp)
+	assert.Equal(t, 7, len(resp.Accounts))
+
+	assertResponseContainsUser(t, resp, getAccount("user1"))
+	assertResponseContainsUser(t, resp, getAccount("user2"))
+
+	// time.Sleep(time.Second * 30)
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:411]: debug line?

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 2]
```diff
@@ -0,0 +1,927 @@
...
+
+	fmt.Println("new Users", newCreatedAccounts)
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:413]: debug line? can it be deleted?

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 3]
```diff
@@ -0,0 +1,927 @@
...
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.GetGroupRequest{Id: "42"}
+	resp, err := cl.GetGroup(context.Background(), req)
+
+	assert.IsType(t, &proto.Group{}, resp)
+	assert.Empty(t, resp)
+	assert.Error(t, err)
+	assert.Equal(t, "{\"id\":\".\",\"code\":404,\"detail\":\"could not read group: open accounts-store/groups/42: no such file or directory\",\"status\":\"Not Found\"}", err.Error())
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:552]: can we unmarshal this string and compare the JSON object?

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 4]
```diff
@@ -0,0 +1,927 @@
...
+	cleanUp(t)
+}
+
+func TestDeleteGroup(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+	grp2 := getTestGroups("grp2")
+	// grp3 := getTestGroups("grp3")
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:559]: what happened to grp3?

File: pkg/proto/v0/accounts.pb.micro_test.go (hunk 3 above)
Comment (dpakach) [reply pkg/proto/v0/accounts.pb.micro_test.go:552]: The server returns this as a string so I don't think that will be really necessary

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 5]
```diff
@@ -0,0 +1,944 @@
...
+	checkError(t, err)
+
+	assert.IsType(t, &proto.Group{}, res)
+
+	// Should return the group but returns empty
+	// assertGroupsSame(t, updatedGroup, res)
+
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, updatedGroup)
+
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:705]: need to check if member was added

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 6]
```diff
@@ -0,0 +1,944 @@
...
+	res, err := cl.RemoveMember(context.Background(), req)
+	checkError(t, err)
+
+	assert.IsType(t, &proto.Group{}, res)
+	//assert.Equal(t, proto.Group{}, *res)
+	// assertGroupsSame(t, grp1, res)
+
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, grp1)
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:813]: need also to check that the member is not in the group anymore

File: pkg/proto/v0/accounts.pb.micro_test.go (hunk 5 above)
Comment (dpakach) [reply pkg/proto/v0/accounts.pb.micro_test.go:705]: the updatedGroups contains the new member too. if the response contains the updated group then I think we can safely assume that them member was added.

File: pkg/proto/v0/accounts.pb.micro_test.go (hunk 6 above)
Comment (dpakach) [reply pkg/proto/v0/accounts.pb.micro_test.go:813]: The original grp1 doesn't contain the member, in the beginning, if the response contains grp1, then we can assume that the member was removed from the group.

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 7]
```diff
@@ -0,0 +1,944 @@
...
+	if result == nil {
+		t.Fatalf("Account not found in response: %s", account.PreferredName)
+	}
+	assertAccountsSame(t, account, result)
+}
+
+func assertResponseNotContainsUser(t *testing.T, response *proto.ListAccountsResponse, account *proto.Account) {
+	for _, a := range response.Accounts {
+		if a.Id == account.Id || a.PreferredName == account.PreferredName {
+			t.Fatal("Account not found in response")
```
Comment (jasson99) [inline pkg/proto/v0/accounts.pb.micro_test.go:230]: account found or not found?

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 8]
```diff
@@ -0,0 +1,944 @@
...
+	if result == nil {
+		t.Fatalf("Group not found in response: %s", group.GetDisplayName())
+	}
+	assertGroupsSame(t, group, result)
+}
+
+func assertResponseNotContainsGroup(t *testing.T, response *proto.ListGroupsResponse, group *proto.Group) {
+	for _, g := range response.Groups {
+		if g.Id == group.Id && g.DisplayName == group.DisplayName {
+			t.Fatal("Group not found in response")
```
Comment (jasson99) [inline pkg/proto/v0/accounts.pb.micro_test.go:262]: Group found or not found?

**Pull request merged**